canonical/maas-ui
# feat: use single filter for machine list actions

## Done

- use single filter for machine list actions
- add machine action status tied to callId `machine.actions = { [callId]: status: string, errors: APIError }`
- add `useDispatchWithCallId`

Note: machine actions will either use `machines` or `selectedFilter` depending on what's passed down - but this will be updated in https://github.com/canonical/app-tribe/issues/1303 to use a single `selectedFilter` for all cases

Will be followed up by: https://github.com/canonical/app-tribe/issues/1393

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine listing
- Select a group of machines
- Go to Take action -> Test
- Verify the button indicates a correct number of machines
- Start tests for machines
- Verify the action has been dispatched with a single called using a filter, e.g. ```action
: "test", filter:  {power_state: ["=unknown"]} ```

not all actions are wired with the new state with `callId` yet  - e.g. abort. It will succeed, but won't show the error message/close on it's own yet.

## Fixes

Fixes: https://github.com/canonical/app-tribe/issues/1286
Fixes: https://github.com/canonical/app-tribe/issues/1392

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.
